api/envoy/extensions/filters/network/redis_proxy/v3/redis_proxy.proto

```diff
@@ -86,6 +87,12 @@ message RedisProxy {
     // original error is passed downstream unchanged. By default, this support is not enabled.
     bool enable_redirection = 3;
 
+    // If ``enable_redirection`` is set to true this option configures the DNS cache that the
+    // connection pool will use to resolve hostnames that are returned with MOVED and ASK responses.
+    // If no configuration is provided, DNS lookups will not be performed (and thus the MOVED/ASK errors
+    // will be propagated verbatim to the user).
+    common.dynamic_forward_proxy.v3.DnsCacheConfig dns_cache_config = 9;
```

Comment:
Can you update the Redis Arch docs to talk about this and maybe have a config example of how to set this up? I would go as far as to say that we should error a config if this is not set since it's broken, but that would be a breaking change for existing users. Can we at least output a warning of what will happen if enable_redirection is set, this is not set, and the server returns MOVED/ASK with host names? I'm mostly foreseeing other people not realizing they have to set this.

Reply:
Yeah I'll update the docs. 

I do think we should emit the warning just once (e.g.: when the filter is instantiated), as opposed to an every MOVED/ASK request since that'd be very spammy.

Reply:
Yeah I meant at config load time once.

Reply:
Ah yes, you did say what _will_ happen. Sorry lack of ☕️.

Reply:
I'll also open an issue to explore enabling the DNS cache by default, once this has been used for a while.